src/event.rs

```diff
@@ -67,6 +67,21 @@ pub enum Event<'a, T: 'static> {
         event: DeviceEvent,
     },
 
+    /// Emitted on macOS to let the application know that a file or files are requested to be opened.
+    ///
+    /// This for example happens when the user double-clicks on a file in Finder and the file's type
+    /// is associated with this application. (Other systems usually pass the path to the files as
+    /// arguments to the program. See `std::env::args()`.)
+    ///
+    /// More precisely this event is emmited when either of `application:openFile:` or
+    /// `application:openFiles:` is called on the application delegate. Winit always responds to
+    /// the OS as if the operation was succesful regardles of wheter this event was handled or
+    /// wheter the files could indeed be opened.
+    ///
+    /// Note that to allow associating file types with an application, said application must have
+    /// an appropriate `Info.plist` file in the application bundle defining `CFBundleDocumentTypes`.
```

Comment:
This is **a lot** of text for such a simple feature. We should aim at being concise with documentation to make sure people are actually able to read it.

I'd advise you to try and shorten it a bit yourself, try to focus on what a user of winit would care about, instead of documenting how the implementation inside of winit works. There's also a few typos, so a quick spellcheck is probably more useful than hoping a reviewer is able to spot everything.

If you need help, please let me know and I can try to make some suggestions.